lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -8310,6 +8327,16 @@ namespace Js
             return false;
         }
 
+     	bool evictCachedType = true;
+        if (DynamicType::Is(type->GetTypeId()))
+        {
+            DynamicType* dynamicType = static_cast<DynamicType*>(type);
+            if ((dynamicType->GetTypeHandler()->IsSimpleDictionaryTypeHandler() || dynamicType->GetTypeHandler()->IsDictionaryTypeHandler()) &&
```

Comment:
Because a type with a path type handler may start off as non-shared but is definitely shareable. Didn't want to evict such types

---
In reply to: [78785136](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78785136) [](ancestors = 78785136)